src/printer.js

```diff
@@ -4798,12 +4799,18 @@ function printAstToDoc(ast, options, addAlignmentSize) {
       return genericPrint(path, options, printGenerically, args);
     }
 
-    return comments.printComments(
+    const withComments = comments.printComments(
       path,
       p => genericPrint(p, options, printGenerically, args),
       options,
       args && args.needsSemi
     );
+
+    if (path.getNode() === options.cursorNode) {
```

Comment:
Ah, that hadn't occurred to me. Fixed in https://github.com/prettier/prettier/pull/1637/commits/d75906d69ae8d38c1c235d822c79c373318e2d79